### PR TITLE
[🔧 esl-carousel]: migration carousel slide `esl-carousel-slide` to Mixin representation

### PR DESCRIPTION
**Pull request by ala-n**
Updates list:
- support for mixin-based slides, new architecture, dependable structure
- a11ty improvements (auto-id, auto-aria)
- fixes for some of nav utils (incomplete state, navigate, initial rendering)
- initial event

### CODE REVIEW

File: src/modules/esl-carousel/plugin/dots/esl-carousel.nav.dots.ts [hunk 1]
```diff
@@ -147,9 +155,9 @@ export class ESLCarouselNavDots extends ESLBaseElement {
   }
 
   /** Updates dots state, rebuilds dots if needed */
-  public update(): void {
+  public update(force?: boolean): void {
     memoize.clear(this, ['count', 'activeIndex']); // invalidate state memoization
-    if (this.$dots.length !== this.count) {
+    if (force || this.$dots.length !== this.count) {
       const $dots = new Array(this.count).fill(null).map((_, index) => this.dotBuilder(index, this));
       $dots.forEach(($dot, index) => $dot.setAttribute('esl-carousel-dot', String(index)));
```
Comment (dshovchko) [inline src/modules/esl-carousel/plugin/dots/esl-carousel.nav.dots.ts:162]: Do we need to set esl-carousel-dot attribute?
It seems to me that this operation performs in the dotBuilder method.

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 2]
```diff
@@ -147,40 +153,33 @@ export class ESLCarousel extends ESLBaseElement {
...
-  public addSlide(slide: HTMLElement | ESLCarouselSlide): void {
+  public addSlide(slide: HTMLElement): void {
     if (!slide) return;
-    if (!(slide instanceof ESLCarouselSlide)) return this.addSlide(ESLCarouselSlide.create(slide));
-    if (slide.parentNode !== this.$slidesArea) {
-      slide.remove();
-      this.$slidesArea?.appendChild(slide);
-    }
-    this.update();
+    slide.setAttribute(this.tagName + '-slide', '');
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.ts:158]: Maybe we should implement getter slideTagName?

File: site/views/examples/carousel.marquee.njk [hunk 3]
```diff
@@ -13,17 +13,17 @@ icon: examples/carousel.svg
...
-          <esl-image mode="cover" lazy
-                        data-alt="{{ 'Carousel slide ' + loop.index }}"
-                        data-src="{{ '/assets/carousel/' + loop.index + '-sm.jpg'  | url }}"></esl-image>
-        </div>
-        </esl-carousel-slide>
+        <li esl-carousel-slide>
+          <div class="img-container">
+            <esl-image mode="cover" lazy
+                          data-alt="{{ 'Carousel slide ' + loop.index }}"
+                          data-src="{{ '/assets/carousel/' + loop.index + '-sm.jpg'  | url }}"></esl-image>
```
Comment (dshovchko) [inline site/views/examples/carousel.marquee.njk:22]: cosmetic: wrong identation

**Review by dshovchko — APPROVED** [review]

File: src/modules/esl-carousel/core/esl-carousel.slide.ts [hunk 4]
```diff
@@ -12,41 +11,54 @@ import type {ESLCarousel} from './esl-carousel';
...
-  @boolAttr() public prev: boolean;
+  /** @returns slide index. */
+  public get index(): number {
+    return this.$carousel!.indexOf(this.$host);
+  }
+  /** @returns whether the slide is active */
+  public get active(): boolean {
+    return this.$carousel!.isActive(this.$host);
+  }
+  /** @returns whether the slide is active */
```
Comment (NastaLeo) [inline src/modules/esl-carousel/core/esl-carousel.slide.ts:29]:
```suggestion
  /** @returns whether the slide is pre-active */
```

File: src/modules/esl-carousel/core/esl-carousel.slide.ts [hunk 5]
```diff
@@ -12,41 +11,54 @@ import type {ESLCarousel} from './esl-carousel';
...
+  public get preActive(): boolean {
+    return this.$carousel!.isPreActive(this.$host);
+  }
+  /** @returns whether the slide is active */
```
Comment (NastaLeo) [inline src/modules/esl-carousel/core/esl-carousel.slide.ts:33]:
```suggestion
  /** @returns whether the slide is next in navigation */
```

File: src/modules/esl-carousel/core/esl-carousel.slide.ts [hunk 6]
```diff
@@ -12,41 +11,54 @@ import type {ESLCarousel} from './esl-carousel';
...
+  public get next(): boolean {
+    return this.$carousel!.isNext(this.$host);
+  }
+  /** @returns whether the slide is active */
```
Comment (NastaLeo) [inline src/modules/esl-carousel/core/esl-carousel.slide.ts:37]:
```suggestion
  /** @returns whether the slide is previous in navigation*/
```

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-carousel/plugin/dots/esl-carousel.nav.dots.ts (hunk 1 above)
Comment (ala-n) [reply src/modules/esl-carousel/plugin/dots/esl-carousel.nav.dots.ts:165]: I've added a comment here, and remove it from default builder
As we agreed: the `esl-carousel-dot` is mandatory system attribute, we do not bother consumer with it but instead making sure it is always presented. (The value of that attribute is also system so user should not rewrite it until he rewrites dots plugin in general)

**Review by abarmina — APPROVED** [review]

**Pull request merged**